src/Core/Resolvers/QueryExecutor.cs

```diff
@@ -805,5 +853,23 @@ private void ValidateSize(long availableSizeBytes, long sizeToBeReadBytes)
                     subStatusCode: DataApiBuilderException.SubStatusCodes.ErrorProcessingData);
             }
         }
+
+        internal virtual void AddDbExecutionTimeToMiddlewareContext(long time)
+        {
+            HttpContext? httpContext = HttpContextAccessor.HttpContext;
+            if (httpContext != null)
+            {
+                lock (_httpContextLock)
```

Comment:
NIT: we should probably add some code comments to explain need of lock for future reference

Reply:
added